go/apps/api/routes/v2_keys_verify_key/handler.go

```diff
@@ -0,0 +1,261 @@
+		Name:        ptr.P(key.Key.Name.String),
+		Permissions: ptr.P(key.Permissions),
+		Roles:       ptr.P(key.Roles),
+		KeyId:       ptr.P(key.Key.ID),
+
+		Credits:    nil,
+		Expires:    nil,
+		Identity:   nil,
+		Meta:       nil,
+		Ratelimits: nil,
+	}
+
+	remaining := key.Key.RemainingRequests
+	if remaining.Valid {
+		res.Data.Credits = ptr.P(remaining.Int32)
+	}
+
+	if key.Key.Expires.Valid {
+		res.Data.Expires = ptr.P(key.Key.Expires.Time.UnixMilli())
+	}
+
+	if key.Key.Meta.Valid {
+		err = json.Unmarshal([]byte(key.Key.Meta.String), &res.Data.Meta)
+		if err != nil {
+			return fault.Wrap(err, fault.Code(codes.App.Internal.UnexpectedError.URN()),
+				fault.Internal("unable to unmarshal key meta"),
+				fault.Public("We encountered an error while trying to unmarshal the key meta data."),
+			)
+		}
+	}
+
+	if key.Key.IdentityID.Valid {
+		res.Data.Identity = &openapi.Identity{
+			ExternalId: key.Key.ExternalID.String,
+			Id:         key.Key.IdentityID.String,
+			Ratelimits: nil,
+			Meta:       nil,
+		}
+
+		for _, ratelimit := range key.GetRatelimitConfigs() {
+			if ratelimit.IdentityID == "" {
+				continue
+			}
+
+			res.Data.Identity.Ratelimits = append(res.Data.Identity.Ratelimits, openapi.RatelimitResponse{
+				AutoApply: ratelimit.AutoApply == 1,
+				Duration:  int64(ratelimit.Duration),
+				Id:        ratelimit.ID,
+				Limit:     int64(ratelimit.Limit),
+				Name:      ratelimit.Name,
+			})
+		}
+
+		if len(key.Key.IdentityMeta) > 0 {
+			err = json.Unmarshal(key.Key.IdentityMeta, &res.Data.Identity.Meta)
+			if err != nil {
+				return fault.Wrap(err, fault.Code(codes.App.Internal.UnexpectedError.URN()),
+					fault.Internal("unable to unmarshal identity meta"),
+					fault.Public("We encountered an error while trying to unmarshal the identity meta data."),
+				)
+			}
+		}
+	}
+
+	if len(key.RatelimitResults) > 0 {
+		ratelimitResponse := make([]openapi.VerifyKeyRatelimitData, 0)
+		for _, result := range key.RatelimitResults {
+			if result.Response == nil {
+				continue
+			}
+
+			exceeded := !result.Response.Success
+			ratelimitResponse = append(ratelimitResponse, openapi.VerifyKeyRatelimitData{
+				AutoApply: result.AutoApply,
+				Duration:  result.Duration.Milliseconds(),
+				Exceeded:  &exceeded,
+				Id:        result.Name,
+				Limit:     result.Limit,
+				Name:      result.Name,
+				Remaining: result.Response.Remaining,
+				Reset:     result.Response.Reset.UnixMilli(),
+			})
+		}
+
+		if len(ratelimitResponse) > 0 {
+			res.Data.Ratelimits = ptr.P(ratelimitResponse)
+		}
+	}
+
+	return s.JSON(http.StatusOK, res)
+}
+
+// convertPermissionsToQuery converts OpenAPI permissions to rbac.PermissionQuery
+func convertPermissionsToQuery(permissions openapi.V2KeysVerifyKeyRequestBody_Permissions) (rbac.PermissionQuery, error) {
+	// Try to unmarshal as string first (single permission)
+	if perm, err := permissions.AsV2KeysVerifyKeyRequestBodyPermissions0(); err == nil {
```

Comment:
surely we can come up with a better name than appending 0 and 1 to these. as a dev I have no idea what the difference is

Reply:
Thats from the openapi generated spec, we can propably try to generate it somewhat different. 

Did we want to allow for recursive permissions like in the v1 api? 
```json
  "permissions": {
    "and": [
      {
        "or": [
          "test"
        ]
      },
      "test2",
      {
        "and": [
          "test123"
        ]
      }
    ]
  }
```
The current openapi spec only allows for either a string or an and/or object

Reply:
I think infinite is causing too many issues with generation.
I wonder if we should instead support a single string and parse it? like `"perm1 or perm2"` or some other form...

Reply:
That sounds like you can write wrong expressions really quickly + it wouldn't be the best dx in the sdks right since it would just be a string?

Reply:
yeah, just a thought really
it's probably not a good idea :D

Reply:
I think we let it stay as is for now and revisit if we can't generate it nicer later?